perl-package/AI-MXNet/lib/AI/MXNet/KVStore.pm

```diff
@@ -244,15 +244,14 @@ method pull(
 method set_optimizer(AI::MXNet::Optimizer $optimizer)
 {
     my $is_worker = check_call(AI::MXNetCAPI::KVStoreIsWorkerNode());
-    if($self->type eq 'dist' and $is_worker)
+    if($self->type =~ /dist/ and $is_worker)
     {
         my $optim_str = MIME::Base64::encode_base64(Storable::freeze($optimizer), "");
         $self->_send_command_to_servers(0, $optim_str);
     }
     else
     {
-        $self->_updater(AI::MXNet::Optimizer->get_updater($optimizer));
```

Comment:
will it work if this line left as is and 254 is $self->_set_updater($self->_updater) otherwise it seems like ->_updater is left undefined.
It's very weird that this is needed, the only thing this patch adds is the removal of the outer sub around the call to the updater. Do we really have stumbled upon proper bug in perl itself ?